pkg/controller/database/cockroach/cockroach_test.go

```diff
@@ -165,15 +180,19 @@ func rookCockroachCluster(im ...rookCockroachClusterModifier) *rookv1alpha1.Clus
 	return i
 }
 
-var _ resource.ExternalClient = &cockroachExternal{}
-var _ resource.ExternalConnecter = &cockroachConnecter{}
+var _ resource.ExternalClient = &external{}
+var _ resource.ExternalConnecter = &connecter{}
 
 func TestConnectCockroach(t *testing.T) {
 	provider := kubev1alpha1.Provider{
 		ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: providerName},
 		Spec: kubev1alpha1.ProviderSpec{
-			Secret: corev1.LocalObjectReference{
-				Name: providerSecretName,
+			Secret: runtimev1alpha1.SecretKeySelector{
+				SecretReference: runtimev1alpha1.SecretReference{
+					Name:      providerSecretName,
+					Namespace: providerSecretName + "space",
```

Comment:
Hurr.

Reply:
Haha was just for the draft I promise :wink: